src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -4922,17 +4915,24 @@ void handleMemberInitializer(IMemberInitializerOperation memberInitializer)
                         _evalStack.Push(Visit(arrayReference.ArrayReference));
                         return (success: true, indicies);
 
-                    case OperationKind.DynamicIndexerAccess: // PROTOTYPE(dataflow): For now handle as invalid case to enable other test scenarios
-                    case OperationKind.None: // PROTOTYPE(dataflow): For now handle as invalid case to enable other test scenarios.
-                                             //                      see Microsoft.CodeAnalysis.CSharp.UnitTests.IOperationTests.ObjectCreationWithDynamicMemberInitializer_01
-                    case OperationKind.Invalid:
-                        return (success: false, arguments: ImmutableArray<IOperation>.Empty);
+                    case OperationKind.DynamicIndexerAccess:
+                        var dynamicIndexer = (IDynamicIndexerAccessOperation)instance;
+                        ImmutableArray<IOperation> arguments = dynamicIndexer.Arguments.SelectAsArray(argument =>
+                        {
+                            int captureId = VisitAndCapture(argument);
+                            return (IOperation)GetCaptureReference(captureId, argument);
+                        });
+                        _evalStack.Push(Visit(dynamicIndexer.Operation));
+                        return (success: true, arguments);
```

Comment:
>arguments [](start = 47, length = 9)

I assume you confirmed the order of evaluation by examining IL.

Reply:
Yes

---
In reply to: [193583423](https://github.com/dotnet/roslyn/pull/27519#discussion_r193583423) [](ancestors = 193583423)